internal/typeinfo/arginfo_test.go

```diff
@@ -137,6 +137,76 @@ func (s *typeInfoSuite) TestArgInfoMap(c *C) {
 	c.Check(kind, DeepEquals, reflect.Map)
 }
 
+func (s *typeInfoSuite) TestArgInfoEmbeddedStruct(c *C) {
+	type EmbeddedString string
+	type StructWithTag struct{}
+	type Embedded3 struct {
+		F3 int `db:"col3"`
+	}
+	type Embedded2 struct {
+		F2 int `db:"col2"`
+		Embedded3
+	}
+	type Embedded1 struct {
+		F1 int `db:"col1"`
+	}
+	type Embeddings struct {
+		EmbeddedString
+		StructWithTag `db:"col4"`
```

Comment:
What happens with such cases?
- I assume select x.* just causes the SQL to run but fail on _get_ if the struct will not hold the return.
- Attempting to insert would fail during parameter binding for incompatible values.

Reply:
If the struct is tagged then it will be passed straight to the driver, this means it must implement the Valuer/Scanner for inputs/outputs respectively.

A comment has been added in the `getStructFields` function to explain this behavior.

Reply:
Also added a tagged field in the struct in this test case to verify that it is ignored by SQLair